rpc_sidecar/src/node_client.rs

```diff
@@ -982,6 +997,18 @@ impl FramedNodeClient {
         observe_reconnect_time(disconnected_start.elapsed());
         Ok(stream)
     }
+
+    async fn reconnect_without_retries(
```

Comment:
I wonder if we could not have a separate set of functions here but instead pass a config that has `max_attempts` set to Finite(1)?